### DISCUSSION TITLE
Can I Automatically Generate Screens from a Column of Form Data?

### DISCUSSION THREAD

**Discussion by sarah-strawberries** [Budibase/budibase#7125] [Help]
So, say I had a form with a column labeled "Type of Fruit" and had several rows in the form so that the "Type of Fruit" column contained "Apple," "Banana," "Orange," etc...  Could I then take that row data and use it to automatically generate a bunch of screens? (E.g. .../app/Banana ) I haven't been able to find a way to do that yet.

**Comment by rochesterj** [comment]
Although there is no way to generate screens (at least that I'm aware of) you can create a screen that gets the "fruit" variable from the URL and handles it.

For example, you can create a new screen at app/fruits/:fruit (this is the screen route)

This means that you have access to the variable {{ URL.fruit }} that contains whatever is in your URL. You can hide/show components (if fruit = banana show this, if apple show this and so on).

Letr me know if this helps you, if it doesn't, let's find another way to do it :)

Thanks!

**Reply by sarah-strawberries** [reply]
Thanks for your feedback! :)

Hmmm...using this method, I would still have to manually add the components for each fruit to the screen and then configure them to show at the appropriate time, correct?

I'm trying to build a directory of sorts starting from a screen with cards that are generated from table data. Each card would be hyperlinked to a screen showing the details of that row entry, and the url would be derived from the row's input for one of the form columns. I've got the cards all set up, and they redirect to the correct urls, but I don't know of a way to make sure that the screens are generated automatically so that the urls actually work. The idea was to have the process of creating screens for individual rows be as automated as possible so I don't have to go back and manually add stuff each time a new row of data is added. Not sure whether there would be a good way in Budibase to do this or not.

**Comment by joebudi** [comment]
Hey Sarah. Could you add some screenshots to illustrate your issue/query?

This will help me work out the solution.

**Reply by sarah-strawberries** [reply]
Yes, I can do that. :)
 
<img width="1558" alt="image" src="https://user-images.githubusercontent.com/75397080/185002588-6707a023-c3f6-4588-b094-e23181b9f0a7.png">
So, basically I have this form that people can fill out to submit the information that they want to share in an online photo directory with contact information for families in a specific community.

Once the information is submitted, a new card will be generated on an index/full directory screen. The card will get its details from the submitted form data, and each previously submitted row will have an associated card. Here's a screenshot showing what this might look like after two people had filled out the form (John Doe for himself and his wife Jane, and Al Martin as an individual): 
<img width="522" alt="image" src="https://user-images.githubusercontent.com/75397080/185237496-b0d7e353-e5aa-403d-a585-398033f37010.png">
(Don't mind the stick figures; you can replace them with professionally done family photos in your mind 😉)

Each of these cards when clicked should change to a screen for that individual row of data, e.g. showing all the details for the family with that last name. Here is what one such screen could look like.
<img width="550" alt="image" src="https://user-images.githubusercontent.com/75397080/185242436-04087826-fb90-4037-b580-9cf2eefd53dc.png">

Again, I did figure out how to get the cards automatically generated, but it's automatically generating the screens corresponding to each card that I haven't figured out quite yet. If there is just not a way to automate this, that is okay and I can do it manually. But saving the effort doing that would be much preferred, if there is a way to do that.

Thanks for your help; I look forward to hearing any ideas you may have!

**Comment by sarah-strawberries** [comment]
I also considered doing something like what they did here:https://github.com/Budibase/budibase/discussions/1829 But that wouldn't have the advantage of the index screen for navigation, at least unless there was some way I could have a variable key in the url for the screens that could be linked to from the cards, something like the solution @rochesterj mentioned...🤔 But then from there, the tricky part would be only showing the data for that specific row, since I'm not sure how I'd go about doing that automatically rather than having to manually add a bunch of conditions to a bunch of components.

**Reply by rochesterj** [reply]
Hi @sarah-strawberries

Thanks for sending these screenshots and the explanation.

It looks like what I've suggested should work fine for you.

You just need a condition in your data provider to filter out the current card, like dynamic filters in a table. 

Let's say you choose to use the entry ID in your url, so you'll create a screen with the route /entry/:id, for example. 

This automatically tells budibase "hey, store anything that comes after /entry/ in a variable called id" . 

Now in your data provider, you'll simply add a filter for entry id = {{ URL.id }} 

Then you use a repeater inside of this data provider and you'll be able to access all your entry properties right away. 

I'm on my phone right now (it's quite late around here!), but if you want I can send you some screenshots to show how you can do it tomorrow. 

Cheers!
